overlord/ifacestate/handlers.go

```diff
@@ -370,36 +368,40 @@ func (m *InterfaceManager) doConnect(task *state.Task, _ *tomb.Tomb) error {
 	if slot == nil {
 		return fmt.Errorf("snap %q has no %q slot", connRef.SlotRef.Snap, connRef.SlotRef.Name)
 	}
-	var slotDecl *asserts.SnapDeclaration
-	if slot.Snap.SnapID != "" {
-		var err error
-		slotDecl, err = assertstate.SnapDeclaration(st, slot.Snap.SnapID)
-		if err != nil {
-			return fmt.Errorf("cannot find snap declaration for %q: %v", slot.Snap.Name(), err)
+
+	// FIXME: check auto connect policy here instead when interface hooks land.
```

Comment:
Is this comment right?  Isn't the policy being checked by doAutoConnect below?

Reply:
doAutoConnect checks policy indeed, but this is too early. Once interface hooks land we need to check auto connect policy after we have all the dynamic attributes from the hooks - in doConnect, around this FIXME comment.

Reply:
I've adjusted this comment per our today's discussion about autoconnect. The existing early autoconnect policy check will remain, second check will be done here (in the hooks PR).